pkg/v1/remote/index.go

```diff
@@ -182,3 +182,48 @@ func (r *remoteIndex) childDescriptor(child v1.Descriptor, platform v1.Platform)
 		platform:   platform,
 	}, nil
 }
+
+// matchesPlatform checks if the given platform can run the required platform.
```

Comment:
maybe say this

// matchesPlatform checks if the given platform matches the required platforms.
// The given platform matches the required platform if
//   - architecture and OS are identical
//   - OS version and variant are identical if provided
//   - features and OS features of the required platform is a subset of those of the given platform

Reply:
- features and OS features of the required platform are subsets of those of the given platform